### PR TITLE
inso cli scripting first pass

### PR DESCRIPTION
**Pull request by jackkav**
motivation: incrementally extend cli features to match app

aim: add the simplest possible implementation which will allow a header to be added by a pre request script, to be confirmed with a test which will rely upon on the script modifying the request

- [x] proof of concept, copied from hidden-window.ts and wired up
- [x] make requireInterceptor work, and use same abstraction as hidden-window
- [x] test case for add header
- [x] test case for require
- [x] test case for send request

reviewer note

consider running bundle tests against build
consider refactoring opportunities
theres a high chance of future inconsistency with hidden window implementation

future work

- post request
- abstract into common implementation?
- remove hidden-window implementation? could be a heavy lift due to async implementation, perhaps not for @ihexxa 

closes INS-4144

### CODE REVIEW

**Review by filfreire — DISMISSED** [review]

**Review by filfreire — DISMISSED** [review]

File: packages/insomnia-inso/src/cli.test.ts [hunk 1]
```diff
@@ -49,51 +49,104 @@ const shouldReturnErrorCode = [
...
 
 describe('inso dev bundle', () => {
-  it.each(shouldReturnSuccessCode)('exit code should be 0: %p', async input => {
-    const result = await runCliFromRoot(input);
-    if (result.code !== 0) {
-      console.log(result);
-    }
-    expect(result.code).toBe(0);
+  describe('exit codes are consistent', () => {
+    it.each(shouldReturnSuccessCode)('exit code should be 0: %p', async input => {
```
Comment (ihexxa) [inline packages/insomnia-inso/src/cli.test.ts:53]: 👍 Table driven test looks pretty neat.

**Comment by jackkav** [comment]
The major contention of this pass
- chance of drift in hidden window context implementation
- slightly different execution context due to the promise handling

My preference would be to use matching execution contexts so that cli task execution results would always match the UI.
That would mean removing the hidden window executor and using a nodejs executor for both.

File: packages/insomnia/src/hidden-window.ts [hunk 2]
```diff
@@ -28,6 +28,8 @@ window.bridge.onmessage(async (data, callback) => {
   }
 });
 
+// This function is duplicated in scriptExecutor.ts to run in nodejs
```
Comment (ihexxa) [inline packages/insomnia/src/hidden-window.ts:31]: I would propose to consider this after the executing mechanism in the CLI side has been proved working well.

**Review by ihexxa — COMMENTED** [review]
The first seems working and just have some minor comments.

**Review by filfreire — APPROVED** [review]

**Pull request merged**